BuildResidentialHPXML/resources/geometry.rb

```diff
@@ -712,9 +712,10 @@ def self.create_single_family_detached(runner:,
   end
 
   def self.has_same_vertices(surface1, surface2)
-    if getSurfaceXValues([surface1]) == getSurfaceXValues([surface2]) &&
-       getSurfaceYValues([surface1]) == getSurfaceYValues([surface2]) &&
-       getSurfaceZValues([surface1]) == getSurfaceZValues([surface2])
+    if getSurfaceXValues([surface1]).sort == getSurfaceXValues([surface2]).sort &&
+       getSurfaceYValues([surface1]).sort == getSurfaceYValues([surface2]).sort &&
+       getSurfaceZValues([surface1]).sort == getSurfaceZValues([surface2]).sort &&
+       surface1.space.get.zOrigin.round(5) == surface2.space.get.zOrigin.round(5)
```

Comment:
Adds z coordinate matching of the surfaces' spaces (to prevent 2nd story walls from being matched with 1st story walls) and incorporates the `.sort` here.

Reply:
@shorowit Is this the actual bugfix?

Reply:
Yes